core/src/main/java/org/web3j/tx/Contract.java

```diff
@@ -380,4 +383,31 @@ protected EventValues extractEventParameters(
 
         return values;
     }
+
+    /**
+     * Subclasses should implement this method to return pre-existing addresses for deployed
+     * contracts.
+     *
+     * @param networkId the network id, for example "1" for the main-net, "3" for ropsten, etc.
```

Comment:
You could put a link to the org.web3j.tx.ChainId class in this Javadoc

Reply:
I actually searched the code base for ropsten, but I think I forget to make it a case insensitive search.  Hadn't seen ChainId.  Curious why not just make that an enum?

Also, ChainId values are bytes, but the max value of a byte is 255 (unsigned).  And the doc referenced actually shows 1337 as one of the recommended ids.  

Shall we change ChainId to an enum and make the values int or, better, long ?

Reply:
The reason for using a byte, is to support [EIP-155](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md), where you place a byte value in the raw transaction object - see [here](https://github.com/web3j/web3j/blob/master/crypto/src/main/java/org/web3j/crypto/TransactionEncoder.java#L39).

That being said further down the discussion in the original [EIP](https://github.com/ethereum/EIPs/issues/155) it is mentioned that the v value when the transaction is RLP encoded shouldn't be limited to a single byte. So web3j could be enhanced to accomodate this. I've created a separate [issue](https://github.com/web3j/web3j/issues/234) for this.